apps/web/src/services/iam/store/user-group-page-store.ts

```diff
@@ -88,14 +83,50 @@ export const useUserGroupPageStore = defineStore('page-user-group', () => {
             state.pageStart = 1;
             state.pageLimit = 15;
             state.searchFilters = [];
+            state.users = {
+                list: [],
+                pageStart: 1,
+                pageLimit: 15,
+                totalCount: 0,
+                selectedIndices: [],
+                searchFilters: [],
+            };
             state.modal = {
                 type: '',
                 title: '',
                 themeColor: 'primary',
             };
         },
-        async listUserGroups() {
-            //     TODO: api connect
+        async listUserGroups(params: UserGroupListParameters) {
+            try {
+                const response = await SpaceConnector.clientV2.identity.userGroup.list<UserGroupListParameters, ListResponse<UserGroupModel>>(params);
+                state.userGroups = response.results || [];
+                state.selectedIndices = [];
+            } catch (e) {
+                ErrorHandler.handleError(e);
+                state.userGroups = [];
+                state.totalCount = 0;
```

Comment:
🤔: Does the data handled differ based on success or failure?

Reply:
I thought that if the list data is not imported, it should be marked as empty array.